main.py

```diff
@@ -65,83 +97,120 @@ def aggregate_model(
                 aggregated_model_weights[key] += user_model_state[key] * (1 / n_peers)
 
     if aggregated_model_weights:
-        print(f"Aggregated models from {aggregated_peers}")
         global_model.load_state_dict(aggregated_model_weights)
-        torch.save(global_model.state_dict(), str(global_model_path))
-        return global_model
+        torch.save(global_model.state_dict(), str(model_output_path))
+        return global_model, missing_peers
     else:
-        return None
-
-
-def network_participants(datasite_path: Path, participants_json_file_path: Path):
-    exclude_dir = ["apps", ".syft"]
-    entries = os.listdir(datasite_path)
-    with open(participants_json_file_path, "r") as f:
-        participants = json.load(f)
-        participants = participants["participants"]
-
-    all_users = []
-    for entry in entries:
-        user_path = Path(datasite_path / entry)
-        is_excluded_dir = entry in exclude_dir
-        is_valid_peer = user_path.is_dir() and not is_excluded_dir
-        if is_valid_peer:
-            all_users.append(entry)
-
-    participants = list(set(participants) & set(all_users))
-    missing_participants = list(set(participants) - set(all_users))
-    print(f"Pretrained model aggregator participants: {participants}")
-    return participants
+        return None, None
+
+
+def advance_pretrained_aggregator(client: Client) -> None:
```

Comment:
Can we rename this to something more intuitive `perform_model_weight_aggregation` ?